demos/control_loop_examples/scripts/robot_kinematics_control_loop.cpp

```diff
@@ -38,39 +38,44 @@ class DummyRobotInterface {
     QPInverseVelocityParameters parameters;
     parameters.dt = dt;
     // apply the inverse velocity
-    JointVelocities desired_joint_velocities = this->robot_.inverse_velocity(desired_eef_twist,
-                                                                             this->joint_positions,
-                                                                             parameters);
+    JointVelocities desired_joint_velocities = this->robot_.inverse_velocity(
+        desired_eef_twist, this->joint_positions, parameters
+    );
     // integrate the new position
     this->joint_positions = dt * desired_joint_velocities + this->joint_positions;
   }
 };
 
-void control_loop_step(DummyRobotInterface& robot,
-                       const Linear<CartesianState>& ds,
-                       const std::chrono::nanoseconds& dt) {
+void control_loop_step(
+    DummyRobotInterface& robot, const std::shared_ptr<IDynamicalSystem<CartesianState>>& ds,
+    const std::chrono::nanoseconds& dt
+) {
   // read the robot state
   robot.read_robot_state();
   // print the state and eef pose
   std::cout << robot.joint_positions << std::endl;
   std::cout << robot.eef_pose << std::endl;
   // get the twist evaluated at current pose
-  CartesianTwist desired_twist = ds.evaluate(robot.eef_pose);
+  CartesianTwist desired_twist = ds->evaluate(robot.eef_pose);
   // send the desired twist to the robot
   robot.send_control_command(desired_twist, dt);
 }
 
 void control_loop(DummyRobotInterface& robot, const std::chrono::nanoseconds& dt, double tolerance) {
-  // set a desired target for the robot eef and a linear ds toward the target
+  // set a desired target for the robot eef and a point attractor ds toward the target
   CartesianPose target(robot.eef_pose.get_name(), robot.eef_pose.get_reference_frame());
   target.set_position(.5, .0, .75);
   Eigen::Quaterniond orientation(Eigen::AngleAxisd(M_PI, Eigen::Vector3d::UnitY()));
   target.set_orientation(orientation);
-  Linear<CartesianState> linear_ds(target);
+  std::shared_ptr<IDynamicalSystem<CartesianState>>
```

Comment:
in most cases I would prefer `auto` here instead of rewriting the type. Since it's a demo there's some benefit to explicitly showing that it's a shared pointer, though at the same time it's very verbose. I just wanted to point that out, but you don't have to change it.

Reply:
I had the exact same thoughts when I did it. Left it there for the sake of clarity